### ISSUE TITLE
Report Bug： dataset can't be feed

### ISSUE THREAD

**Issue by xubenben** [OlavHN/attention-over-attention#2]
Dear OlavHN:

As I have reported that the 

> "python model.py --training=False --name=my_model --epochs=1 --dropout_keep_prob=1" 

run too long. I find that that's due to the failure of feed_dict, which means that the dataset variable is always the default value.

If we change "dataset = tf.placeholder_with_default(0, [])" to "dataset = tf.placeholde(tf.int32, [])", the code will be wrong, with 

> message:"You must feed a value for placeholder tensor 'Placeholder' with dtype int32".

**Comment by OlavHN** [comment]
I took a look at this now, and you're right. Looks like when you're doing a feed_dict, specifying the queue it only works for the first run of the dequeue operation. The next one will ignore the feed_dict value. This was not how I expected it to work, and **more importantly it means I've run my validation / test runs on the training data!**

Will have to take a closer look at how the feed_dict and queue ops interact. Perhaps giving up the plan of running all queues in the same session. Suggestions and/or pull requests very welcome!

**Comment by OlavHN** [comment]
I quickly ran the same model, forcing the test set, and got an accuracy of 0.70580808080808088 :(

**Comment by xubenben** [comment]
So you mean that the 74.3% performance you have reported isn't precisely？^_^

**Comment by OlavHN** [comment]
Yes, you are right.

When (wrongly) testing the model on the training set I noticed it had very low accuracy through most of the data, then doing a large jump towards the end, implying the model is overfitting on the later data. So I'm hopeful there isn't an implementation bug, but that tuning the regularisation a little will help :)

**Comment by LibraSkywalker** [comment]
I've met the same problem. When I'm trying to revise your code to predict some RC problems, I was amazed that even when I have not labeled the cases, there're still 70+% accuracy, I have barely any idea why such situation happened. Later I found out that the **feed_dict** operation is actually not working.  I don't know if it's because the update of tensorflow cause such problem or a bug, but I fixed it by a really simple and nasty method, that is to initialize dataset differently due to different flags . Any way, I really appreciated your contribution. This project helps me a lot.
